pytket/pytket/qasm/qasm.py

```diff
@@ -335,18 +335,84 @@ def mixedcall(self, tree: List) -> Iterator[CommandDict]:
         except StopIteration:
             args = next_tree
             pars = []
-        params = [f"({par})/pi" for par in pars]
+
+        params = [
+            f"({par})/pi" for par in pars
+        ]  # do we want to do this for all parameters?
+
+        treat_as_barrier = [
+            "sleep",
+            "order2",
+            "order3",
+            "order4",
+            "order5",
+            "order6",
+            "order7",
+            "order8",
+            "order9",
+            "order10",
+            "order11",
+            "order12",
+            "order13",
+            "order14",
+            "order15",
+            "order16",
+            "order17",
+            "order18",
+            "order19",
+            "order20",
+            "group2",
+            "group3",
+            "group4",
+            "group5",
+            "group6",
+            "group7",
+            "group8",
+            "group9",
+            "group10",
+            "group11",
+            "group12",
+            "group13",
+            "group14",
+            "group15",
+            "group16",
+            "group17",
+            "group18",
+            "group19",
+            "group20",
+        ]
+        # other opaque gates, which are not handled as barrier
+        # ["RZZ", "Rxxyyzz", "Rxxyyzz_zphase", "cu", "cp", "rccx", "rc3x", "c3sqrtx"]
+
         if opstr in self.gate_dict:
-            gdef = self.gate_dict[opstr]
-            op: Dict[str, Any] = {"type": "CustomGate"}
-            box = {
-                "type": "CustomGate",
-                "id": str(uuid.uuid4()),
-                "gate": gdef,
-            }
-            box["params"] = params
-            op["box"] = box
-            params = []  # to stop duplication in to op
+            op: Dict[str, Any] = {}
+            if opstr in treat_as_barrier:
+                params = [p[1:-4] for p in params]
```

Comment:
Can you explain this please?

Reply:
Yes, it removes the `(` and `)/pi` added in https://github.com/CQCL/tket/blob/feature/opaque-barrier-gates-qasm/pytket/pytket/qasm/qasm.py#L340

The original `pars` variable is empty after L340. Could also solve this with removing `(` and `)/pi` from the `params` variable and adding a second variable somewhere including this again if you prefer this?

Reply:
How about just not adding the "(" and ")/pi" when opstr is in treat_as_barrier?

Reply:
good point, done.